lib/apytypes/_array_functions.py

```diff
@@ -2167,6 +2169,281 @@ def fullrange(
         raise ValueError("Could not determine array type")
 
 
+@overload
+def export_csv(
+    a: APyFixedArray,
+    fname: str,
+    delimiter: str = ",",
+    layout: str = "default",
+) -> None: ...
+
+
+@overload
+def export_csv(
+    a: APyFloatArray,
+    fname: str,
+    delimiter: str = ",",
+    layout: str = "default",
+) -> None: ...
+
+
+def export_csv(
+    a: APyArray,
+    fname: str,
+    delimiter: str = ",",
+    layout: str = "default",
+) -> None:
+    """
+    Store the bit values of an array to a CSV file.
+
+    The function supports multi-dimensional arrays:
+    - For 1-D arrays, all elements are written on a single line.
+    - For 2-D arrays, each row of the array is written on a separate line.
+    - Higher-dimensional arrays must be handled manually unless supported by the chosen layout.
+
+    By default, a 1-D array will be exported as a single row CSV, and a 2-D array will be exported as multiple rows.
+
+    .. attention::
+        The index order of arrays may differ between APyTypes and other programs.
+
+    .. note::
+
+        The VUnit layout transposes the array so that the index order of VUnit's ``get`` function will behave the same as in APyTypes.
+        For a 2-D array, VUnit's ``integer_array_t`` will have the shape ``height, width = a.shape``.
```

Comment:
It is not possible to link directly to the type, but you can link to the package: 
```
:std:doc:`data_types/integer_array`
```

In that case you need to add `(https://vunit.github.io/, None)` to `intersphinx_mapping` in `conf.py`